scaleway/resource_compute_instance_server_test.go

```diff
@@ -0,0 +1,266 @@
+				),
+			},
+			{
+				Config: testAccCheckScalewayComputeInstanceServerConfigState("standby"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayComputeInstanceServerExists("scaleway_compute_instance_server.base"),
+					resource.TestCheckResourceAttr("scaleway_compute_instance_server.base", "state", "standby"),
+				),
+			},
+			{
+				Config: testAccCheckScalewayComputeInstanceServerConfigState("stopped"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayComputeInstanceServerExists("scaleway_compute_instance_server.base"),
+					resource.TestCheckResourceAttr("scaleway_compute_instance_server.base", "state", "stopped"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccScalewayComputeInstanceServerState2(t *testing.T) {
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckScalewayComputeInstanceServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckScalewayComputeInstanceServerConfigState("stopped"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayComputeInstanceServerExists("scaleway_compute_instance_server.base"),
+					resource.TestCheckResourceAttr("scaleway_compute_instance_server.base", "state", "stopped"),
+				),
+			},
+			{
+				Config: testAccCheckScalewayComputeInstanceServerConfigState("standby"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalewayComputeInstanceServerExists("scaleway_compute_instance_server.base"),
+					resource.TestCheckResourceAttr("scaleway_compute_instance_server.base", "state", "standby"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckScalewayComputeInstanceServerExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("resource not found: %s", n)
+		}
+
+		instanceApi, zone, ID, err := getInstanceAPIWithZoneAndID(testAccProvider.Meta(), rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = instanceApi.GetServer(&instance.GetServerRequest{ServerID: ID, Zone: zone})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckScalewayComputeInstanceServerDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "scaleway_compute_instance_ip" {
+			continue
+		}
+
+		instanceApi, zone, ID, err := getInstanceAPIWithZoneAndID(testAccProvider.Meta(), rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = instanceApi.GetServer(&instance.GetServerRequest{
+			ServerID: ID,
+			Zone:     zone,
+		})
+
+		// If no error resource still exist
+		if err == nil {
+			return fmt.Errorf("Server (%s) still exists", rs.Primary.ID)
+		}
+
+		// Unexpected api error we return it
+		if !is404Error(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckScalewayComputeInstanceServerConfigMinimal() string {
```

Comment:
Might be more logica to put the test cases at the top of the file. And why create a function to return them of we can just put them in a var?

@jerome-quere and I put them at the top of the file, and created arrays for tests that have multiple steps. Look in `resource_compute_instance_ip_test.go` for an example.